lib/node/http.js

```diff
@@ -21,6 +21,8 @@ const Claim = require('../primitives/claim');
 const Address = require('../primitives/address');
 const Network = require('../protocol/network');
 const pkg = require('../pkg');
+const AWS = require('aws-sdk');
```

Comment:
What are some other options besides requiring this package for the whole full node? Not blocking but not ideal since it's a big dependency and would probably never make it into upstream.

Could this be an optional peer dependency or something?

Reply:
I'm not sure how peer dependencies work, exactly. I'd considered adding it as a separate plugin.

Reply:
I believe aws sdk v2 lets you import only the services you are planning to use as well, that would trim down the loaded lib size abit. 

var S3 = require('aws-sdk/clients/s3');

Reply:
> I'm not sure how peer dependencies work, exactly. I'd considered adding it as a separate plugin.

Peer dependency is the wrong word since it means something really specific in node.js/npm. I trust aws-sdk but it feels bad adding a 50mb dependency for this one feature. Do you have a clear idea of how this dependency could live in a plugin so `hsd` stays clean?

Reply:
A plugin would just need access to the `Chain`, which is itself a plugin that it could take a dependency on, I believe. So a dump-zone-to-s3 plugin would pretty much be this code with some boilerplate to create the plugin itself and then we'd need some other way of triggering it than the HTTP endpoint on the `Node` plugin interface.

Since it assumes AWS anyway, we could use a queue. If that's too much we could just put up another HTTP interface on a third port. If we wanted to move more of our custom functionality into plugins, it'd be easy to extend that HTTP interface to cover all of them with a namebase meta-plugin that aggregated the calls

Reply:
Bump @turbomaze @rozaydin

Reply:
Replying in slack